source/reactor/src/EventLoopImpl.cpp

```diff
@@ -0,0 +1,144 @@
+#include "reactor/EventLoopImpl.h"
+
+#include "EventReaction.h"
+#include "TimeTools.h"
+
+#include <algorithm>
+
+namespace u {
+namespace reactor {
+
+bool EventLoopImpl::registerHandlers(int fd, long timeout, const EventHandlers &handlers)
+{
+    if (m_eventReactions.find(fd) == m_eventReactions.end()) {
+        m_eventReactions[fd] = std::make_shared<EventReaction>(fd, timeout, handlers);
+        return true;
+    }
+
+    return false;
+}
+
+bool EventLoopImpl::removeHandlers(int fd) { return m_eventReactions.erase(fd) > 0; }
```

Comment:
Base class here is an interface. By definition, "An interface describes the behavior or capabilities of a C++ class without committing to a particular implementation of that class".

Since the m_eventReactions member is used by a particular implementation of the EventLoop interface, it must be defined in that implementation, not in the interface. 

Also I cannot think of a test where the mock would ever need to *store* installed handlers. I'm sure there will be tests with expectations on which handlers are installed.